### DISCUSSION TITLE
Multi-Locale Editing UI

### DISCUSSION THREAD

**Discussion by zubricks** [payloadcms/payload#4195] [Roadmap]
## Multi-Locale Editing

This feature introduces a multi-locale editing view where you can access and edit multiple locales from a single document view, instead of switching between locales one at a time.

The editor should be able to select which locales to show at any given time, and a field per locale should be rendered allowing the editor to view multiple locales at once.

This feature will also address the following localization enhancements:
- Renovate Status Component in Document View to Show All Locale Statuses - [Feature Request #13437](https://github.com/payloadcms/payload/discussions/13437)
- Option to Display Fallback Data within UI - [Feature Request #13448](https://github.com/payloadcms/payload/discussions/13448)
- Required Fields per Locale - [Feature Request #13445](https://github.com/payloadcms/payload/discussions/13445)

### Status

Design is not finalized yet, and implementation details may evolve as we refine the UX for v4.

---

### 📢 Feedback Welcome

Suggestions, concerns, and additional ideas around multi-locale workflows are welcome in this thread. Community input will help shape the final implementation.

**Comment by matthijs166** [comment]
This would be awesome in combination with this plugin!! https://github.com/r1tsuu/payload-plugin-translator

Can't wait to see this feature release 👍

**Reply by JessRynkar** [reply]
Great plugin! Thanks for sharing

**Comment by justuskilianwolff** [comment]
I'd love that! Do you know roughly when it's gonna make it?

**Comment by Kenji-Nguyen** [comment]
I can't wait for this feature, as this is would significantly reduce the hurdle for our clients where editing multiple locales is a requirement.
Any updates on this feature?

**Comment by supra-dheerajacharya** [comment]
I’m eager for this feature! Any idea when it might become available?

**Comment by mik01aj** [comment]
Hey, I've seen this feature on your homepage (I mean, look at the screenshot), does that mean that it is already available? I'm a bit confused.
![image](https://github.com/user-attachments/assets/c7cac860-d20e-4af1-91ca-9d2bf2e67f7f)

**Comment by christophemenager** [comment]
This feature would be amazing. I am really struggling with translations with only two locales. 
The UX is pretty poor : 
- you can't edit multiple locales at the same time
- you have to click on the smallest button ever on the top right to change the locale
- you can validate a locale but keep required fields not filled on an other local with no error and no way to know that another locale is missing fields

**Comment by anyuj** [comment]
Any update on this? Need this feature badly. 😭

**Comment by snakemastr** [comment]
I would love this. Is this still worked on?

**Comment by anyuj** [comment]
I’d like to gently bump this discussion. As many others have mentioned, localization is at the heart of most modern CMS workflows. Being able to edit and compare multiple locales side by side in the UI would be a true game changer for content teams, and it will dramatically improve efficiency. If there’s any way to raise the priority of this functionality, it would be greatly appreciated by those of us who rely on Payload for multilingual projects.

**Comment by jacktsin1** [comment]
Any updates on this??

**Comment by JessRynkar** [comment]
Thank you all for your patience on this feature and we love to see this level of demand 🙏 

We will be discussing the implementation of this feature in our sprint planning next week, I will report back then with an accurate ETA. For now, hold tight, it will prioritized for the near future!

**Comment by johanneswicki** [comment]
Any updates on this??

**Comment by steveninety** [comment]
I'd love to see the multiple locales render on a per-field basis without duplicate labels and descriptions. Otherwise things could get clunky. Might be difficult to implement given how fields currently render, but everything starts with an idea... 

<img width="1440" height="1555" alt="payloadcms-ui-ux-redesign" src="https://github.com/user-attachments/assets/12e54e0d-9c78-4509-9063-b387d1db98cd" />

**Reply by kirill-mally** [reply]
@steveninety Just wondering – what's the name of the CMS you used for that screenshot? Or is it just a UI mockup idea?

**Reply by steveninety** [reply]
@kirill-mally Just a Figma mockup that I made of how I envisioned the UI for PayloadCMS.

**Comment by rolfsdMediaMarktSaturn** [comment]
To have all the language fields below each other makes it hard to keep an overview about the whole site. 

This view would be much cleaner: 
![screenshot](https://github.com/user-attachments/assets/9265892b-1f0f-432b-ba24-bb7ba00aa2a0)
Source: https://github.com/r1tsuu/payload-enchants/tree/master/packages/better-localized-fields (outdated plugin as its not maintained)

**Reply by cbratschi** [reply]
It depends on the field. Seeing two or more values at once makes it easier to synchronize values.

Custom buttons to copy or translate would be useful too.

**Comment by DanRibbens** [comment]
We will need to keep in mind [Duplicating and Copy/Pasting Blocks rows only copies the active locale](https://github.com/payloadcms/payload/discussions/14515#top) with this work.

**Comment by zedbenaa** [comment]
Any updates please?

**Comment by leonedev** [comment]
News on this? Would be great to use it and it is crucial, I'm struggling with just two locales!

**Comment by axe312ger** [comment]
This would be superb. Especially having multiple locales values visible at the same time would help editors to keep stuff in sync and ease editing pain!


Contentfuls structure is super simple but pretty nice for it. They allow you to enable/disable locales while editing.

![](https://images.ctfassets.net/aeu1amk9b600/24ZWBTFnwCZbvtS0HgMZzH/93e78ac5fa91a099e6ffef339b5291e3/Field_Level_Localization.png?w=1320&q=60&fm=webp)

**Comment by MeltingOpera** [comment]
Any updates on this? This was opened almost 3 years ago and the roadmap shows it as priority 1 ...

**Comment by JessRynkar** [comment]
Hey all — a good chunk of the design and foundational work for multi-locale editing actually got started but unfortunately it got cut in favour of other priorities.

I still believe this is needed and valuable, and the feedback in this thread makes that case pretty well on its own. 

I will push to get this back on the roadmap but cannot promise any timeline, and how prioritisation is handled is beyond my control. 

All the interest here is definitely not forgotten, I'll come back with an update when there's something real to share.

Thanks for sticking with this one.

**Reply by leonedev** [reply]
> Hey all — a good chunk of the design and foundational work for multi-locale editing actually got started but unfortunately it got cut in favour of other priorities.
> 
> 
> 
> I still believe this is needed and valuable, and the feedback in this thread makes that case pretty well on its own. 
> 
> 
> 
> I will push to get this back on the roadmap but cannot promise any timeline, and how prioritisation is handled is beyond my control. 
> 
> 
> 
> All the interest here is definitely not forgotten, I'll come back with an update when there's something real to share.
> 
> 
> 
> Thanks for sticking with this one.

Thanks for the news, but don't forget this please. This feature is really crucial when deciding to use Payload CMS or other ones.
